pkg/crd/kubermatic/v1/preset.go

```diff
@@ -300,11 +300,13 @@ func (s AWS) IsValid() bool {
 type Openstack struct {
 	PresetProvider `json:",inline"`
 
-	Username string `json:"username"`
-	Password string `json:"password"`
-	Tenant   string `json:"tenant"`
-	TenantID string `json:"tenantID"`
-	Domain   string `json:"domain"`
+	Username                    string `json:"username,omitempty"`
```

Comment:
Here, I am not sure. I added `omitempty` because username is not required anymore. It's either username/password or applicationcredentialid/secret. This will be enforced in the validations.